src/commands/scale.rs

```diff
@@ -0,0 +1,423 @@
+                                Ordering::Less => amount.red(),
+                            }
+                            .to_string()
+                        }
+                    ),
+                )
+            })
+            .collect::<Vec<PromptRegion>>();
+        let p = prompt_select_with_cancel("Select a region <esc to finish>", regions)?;
+        if let Some(region) = p {
+            let amount_before = if let Some(updated) = updated.get(&region.0.name) {
+                *updated
+            } else if let Some(previous) = existing.as_object().unwrap().get(&region.0.name) {
+                previous.get("numReplicas").unwrap().as_u64().unwrap()
+            } else {
+                0
+            };
+            let prompted = prompt_u64_with_placeholder_and_validation_and_cancel(
+                format!(
+                    "Enter the amount of replicas for {} <esc to go back>",
+                    region.0.name.clone()
+                )
+                .as_str(),
+                amount_before.to_string().as_str(),
+            )?;
+            if let Some(prompted) = prompted {
+                let parse: u64 = prompted.parse()?;
+                updated.insert(region.0.name.clone(), parse);
+            } else {
+                // esc pressed when entering number, go back to selecting regions
+                continue;
+            }
+        } else {
+            // they pressed esc to cancel
+            break;
+        }
+    }
+    Ok(updated.clone())
+}
+
+async fn update_regions_and_redeploy(
+    configs: Configs,
+    client: reqwest::Client,
+    linked_project: LinkedProject,
+    latest_id: Option<String>,
+    region_data: Value,
+) -> Result<(), anyhow::Error> {
+    let spinner = indicatif::ProgressBar::new_spinner()
+        .with_style(
+            indicatif::ProgressStyle::default_spinner()
+                .tick_chars(TICK_STRING)
+                .template("{spinner:.green} {msg}")?,
+        )
+        .with_message("Updating regions...");
+    spinner.enable_steady_tick(Duration::from_millis(100));
+    post_graphql::<mutations::UpdateRegions, _>(
+        &client,
+        configs.get_backboard(),
+        mutations::update_regions::Variables {
+            environment_id: linked_project.environment,
+            service_id: linked_project.service.unwrap(),
+            multi_region_config: region_data,
+        },
+    )
+    .await?;
+    spinner.finish_with_message("Regions updated");
+    if let Some(latest) = latest_id {
+        let spinner = indicatif::ProgressBar::new_spinner()
+            .with_style(
+                indicatif::ProgressStyle::default_spinner()
+                    .tick_chars(TICK_STRING)
+                    .template("{spinner:.green} {msg}")?,
+            )
+            .with_message("Redeploying...");
+        spinner.enable_steady_tick(Duration::from_millis(100));
+        post_graphql::<mutations::DeploymentRedeploy, _>(
```

Comment:
Instead of redeploying here, I think you should be calling `serviceInstanceDeployV2`. Redeploying will intentionally use the config from the deploy you are redeploying. This is why the updated multi region config is not being picked up. If you use service instance deploy, it will use the current service config and deploy the latest commit/image/up deploy.

![image](https://github.com/user-attachments/assets/747c74ad-75a3-4fd8-bc6e-856f01c95927)